libfaasm/core.cpp

```diff
@@ -181,6 +182,36 @@ unsigned int faasmChain(FaasmFuncPtr funcPtr,
     return __faasm_chain_ptr(funcPtr, inputData, inputDataSize);
 }
 
+unsigned int faasmChainBatch(FaasmFuncPtr funcPtr,
+                             const char* inputData,
+                             int nFuncs)
+{
+    size_t inputDataSize = ::strlen(inputData);
```

Comment:
This function feels a bit odd in a serverless setting, as we are specifying the level of parallelism. I understand it is used to give each one a different piece of input, but ideally this would be done transparently to user code.

That being said, this is quite unrelated to any of the work we are doing, so we can merge it in and think about it later.